internal/pkg/config/config.go

```diff
@@ -284,27 +284,59 @@ func (fm funcMatcher) MatchFunction(path, receiver, name string) bool {
 	return fm.Package.MatchString(path) && fm.Receiver.MatchString(receiver) && fm.Method.MatchString(name)
 }
 
-var readFileOnce sync.Once
-var readConfigCached *Config
-var readConfigCachedErr error
+// readConfigResults reduces disk access across multiple ReadConfig calls.
+type readConfigResults struct {
+	once sync.Once
+	conf *Config
+	err  error
+}
 
-// ReadConfig reads, parses, and validates config file.
-func ReadConfig() (*Config, error) {
-	readFileOnce.Do(func() {
+// configCache safely stores readConfigResults for concurrent access.
+type configCache struct {
+	mu    sync.Mutex
+	cache map[string]*readConfigResults
+}
+
+// Instantiates readConfigResults if file has not yet been loaded
+func (w *configCache) read(file string) (*Config, error) {
+	w.instantiateCacheForFile(file)
+
+	cc := w.cache[file]
+	cc.once.Do(func() {
 		c := new(Config)
 		bytes, err := ioutil.ReadFile(configFile)
```

Comment:
Seems like `configFile` should be `file`?

Reply:
Oof.  Bad overlook and good catch!